src/lib/helpers.ts

```diff
@@ -60,6 +61,39 @@ export function assertIsPlainObject(
   }
 }
 
+export function assertIsValidCodeVerifier(
+  verifier: unknown
+): asserts verifier is string {
+  assertIsString(verifier, "Invalid 'code_verifier' type");
+  if (!isValidPkceCodeVerifier(verifier)) {
+    throw new AssertionError({
+      message:
+        "Invalid 'code_verifier'. The verifier does not confirm with the RFC7636 spec. Ref: https://datatracker.ietf.org/doc/html/rfc7636#section-4.1",
+    });
+  }
+}
+
+export function assertIsCodeChallenge(
+  challenge: CodeChallenge | undefined
+): asserts challenge is CodeChallenge {
+  if (challenge === undefined) {
+    throw new AssertionError({
+      message: 'code_challenge required',
+    });
+  }
+}
```

Comment:
This one also doesn't behave as a type assertion. It should assert that `challenge` is either:

- a `CodeChallenge` (by ensuring that it is an object that has a `challenge` field of type `string`, and a `method` field that is a `PKCEAlgorithm`), or
- `undefined` (but you throw an error if it is).

Looking at how you use it, I have the impression that you don't want that `challenge` is `undefined`, because it would throw a null reference error at line 91 in this same file if it were.

Reply:
That type is kind of only used internally, is it worth typechecking the values then - given how you look at its usage

Reply:
I deleted the assertion method